### PR TITLE
Allow symbol for coupling_mhz in CouplerPulse gate

### PR DESCRIPTION
**Pull request by maffoo**
This is something we use on our internal version of this gate.

### CODE REVIEW

**Review by vtomole — COMMENTED** [review]
If you want to allow the passing of symbolic expressions, you won't be able to evaluate the `repr`

```python
gate = coupler_pulse.CouplerPulse(
        hold_time=cirq.Duration(nanos=10), coupling_mhz=sympy.cos(sympy.Symbol('x')) + 1 , rise_time=cirq.Duration(nanos=18)
    )
    cirq.testing.assert_implements_consistent_protocols(
        gate,
        setup_code='import cirq\nimport numpy as np\nimport sympy\nimport cirq_google',
        qubit_count=2,
        ignore_decompose_to_default_gateset=True,
    )
```

Fails with a  `NameError("name 'cos' is not defined")`.

The resolver doesn't seem to work with this change

```python
circuit = cirq.Circuit(gate.on(cirq.LineQubit(0), cirq.LineQubit(1)))
print(cirq.resolve_parameters(circuit, {"x": 0.1}))
# prints
# 0: ───/‾‾(10 ns@cos(x) + 1MHz)‾‾\───
#       │
# 1: ───/‾‾(10 ns@cos(x) + 1MHz)‾‾\───
```

**Comment by maffoo** [comment]
Thanks for taking a look, @vtomole. I don't think repr evaluate is particularly important and I presume the same thing would fail for other parameterized gates as well. I did add support for parameter resolution and related protocols (`parameter_names` and `is_parameterized`). PTAL

**Review by vtomole — COMMENTED** [review]
> I don't think repr evaluate is particularly important

Not important, but should be as simple as modifying `CouplerPulse`'s `repr` to use `proper_repr`

```python 
from cirq._compat import proper_repr
def __repr__(self) -> str:
        return (
            'cirq_google.experimental.ops.coupler_pulse.'
            + f'CouplerPulse(hold_time={self.hold_time!r}, '
            + f'coupling_mhz={proper_repr(self.coupling_mhz)}, '
            + f'rise_time={self.rise_time!r}, '
            + f'padding_time={self.padding_time!r})'
        )
```

File: cirq-google/cirq_google/experimental/ops/coupler_pulse.py [hunk 1]
```diff
@@ -92,6 +92,32 @@ def __str__(self) -> str:
             + f'padding_time={self.padding_time})'
         )
 
+    def _is_parameterized_(self) -> bool:
+        return (
+            cirq.is_parameterized(self.hold_time)
+            or cirq.is_parameterized(self.rise_time)
+            or cirq.is_parameterized(self.padding_time)
+            or cirq.is_parameterized(self.coupling_mhz)
```
[the remark this answers is not in the document]
Comment (maffoo) [reply cirq-google/cirq_google/experimental/ops/coupler_pulse.py]: Done.

**Review by vtomole — APPROVED** [review]
LGTM

**Pull request merged**